src/application/useCases/LoginUseCase.ts

```diff
@@ -0,0 +1,32 @@
+import { UserRepository } from '@/domain/repositories/UserRepository'
+
+type UserLoginUseCaseResponse = {
+  success: boolean
+  message?: string
+}
```

Comment:
Pra bater com o funcionamento que o front tem atualmente, que era compatível com a api no formato anterior, é bom adicionar no retorno do login o username e um accessToken (principalmente pra poder fazer algo sobre teste de segurança lá) igual o retorno anterior.
```suggestion
type UserLoginUseCaseResponse = {
  success: boolean
  message?: string
  username?: string
  accessToken?: string
}
```